src/lyapunovspectrum.jl

```diff
@@ -86,51 +96,89 @@ function modifiedGS!(A, Q, R, aⱼ, qᵢ, vⱼ)
     return nothing
 end
 
-function variational_eqs!(t::Taylor1{T}, x::Vector{Taylor1{U}},
-        dx::Vector{Taylor1{U}}, jac::Array{Taylor1{U},2}, jacobianfunc! =Nothing) where {T<:Real, U<:Number}
-    # Dimensions of phase-space: dof
-    nx = length(x)
-    dof = size(jac, 1)
-    if jacobianfunc! != Nothing
-        # Stability matrix
-        jacobianfunc!(jac, t, x)
+"""
+    lyap_jetcoeffs!(t, x, dx, jac, varsaux)
+
+Similar to [`jetcoeffs!`](@ref) for the calculation of the Lyapunov spectrum.
+Updates *only* the elements of `x` which correspond to the solution of the 1st-order
+variational equations \$\\dot{\\xi}=J \\cdot \\xi\$, where \$J\$ is the Jacobian
+matrix, i.e., the linearization of the equations of motion. `jac` is the Taylor
+expansion of \$J\$ wrt the independent variable, around the current initial
+condition. `varsaux` is an auxiliary array of type `Array{eltype(jac),3}` to
+avoid allocations. Calling this method assumes that `jac` has been computed
+previously using [`stabilitymatrix!`](@ref).
+
+"""
+function lyap_jetcoeffs!(t::Taylor1{T}, x::AbstractVector{Taylor1{S}},
+        dx::AbstractVector{Taylor1{S}}, jac::Matrix{Taylor1{S}},
+        varsaux::Array{Taylor1{S},3}) where {T <: Real, S <: Number}
+    order = t.order
+    # `dofrange` behaves like 1:dof, where `dof = size(jac, 1)`. Used to initialize `b` and `I`
+    dofrange = axes(jac, 1)
+    # The `view` below allows us to obtain CartesianIndices from `varsaux` when calling `eachindex(b)`
+    # `varsaux` is an array with size n×n×n
+    b = view(varsaux, dofrange, dofrange, dofrange)
+    # We use `I` to map between cartesian and linear indices
+    I = LinearIndices((dofrange, dofrange))
```

Comment:
I think `I` is not the best name, since `LinearAlgebra` exports `I` which is the identity matrix; I thus suggest to change the name of that variable.